documentation/library_guide/kernel_templates/single_pass_scan.rst

```diff
@@ -0,0 +1,213 @@
+| - ``in_rng``, ``out_rng`` (2).                 |                                                                     |
+|                                                | - :ref:`USM pointers <use-usm>` (1),                                |
+|                                                | - :ref:`oneapi::dpl::begin and oneapi::dpl::end                     |
+|                                                |   <use-buffer-wrappers>` (1),                                       |
+|                                                | - ``sycl::buffer`` (2),                                             |
+|                                                | - :ref:`oneapi::dpl::experimental::ranges::views::all               |
+|                                                |   <viewable-ranges>` (2),                                           |
+|                                                | - :ref:`oneapi::dpl::experimental::ranges::views::subrange          |
+|                                                |   <viewable-ranges>` (2).                                           |
+|                                                |                                                                     |
++------------------------------------------------+---------------------------------------------------------------------+
+| ``binary_op``                                  | A function object that is applied to the elements of the input.     |
+|                                                |                                                                     |
++------------------------------------------------+---------------------------------------------------------------------+
+| ``param``                                      | A :doc:`kernel_param <../kernel_configuration>` object.             |
+|                                                |                                                                     |
++------------------------------------------------+---------------------------------------------------------------------+
+
+
+**Type Requirements**:
+
+- The element type of sequence to scan must be a 32-bit or 64-bit bit C++ integral or floating-point type.
+- The result is non-deterministic if the binary operator is non-associative (such as in floating-point addition)
+  or non-commutative.
+
+
+.. note::
+
+  Current limitations:
+
+  - The function will internally block until the issued kernels have completed execution.
+    Although intended in the future to be an asynchronous call, the algorithm is currently synchronous.
+  - The SYCL device associated with the provided queue must support 64-bit atomic operations if the element type is 64-bits.
+  - The binary operator must have a known identity. That is, ``sycl::has_known_identity<Op>::value`` must
+    evaluate to true. Such operators are listed in the `SYCL 2020 specification <https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#table.identities>`_.
```

Comment:
`has_known_identity` requires accumulator type and operation type. Let's adjust the description accordingly: 

```suggestion
  - There must be a known identity value for the provided combination of the element type and the binary operation. That is, ``sycl::has_known_identity_v`` must evaluate to true. Such operators are listed in the `SYCL 2020 specification <https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#table.identities>`_.
```